### PR TITLE
reducing the poll times

### PR DESCRIPTION
**Pull request by bpramod**

### CODE REVIEW

File: VMBackup/main/PluginHost.py [hunk 1]
```diff
@@ -199,8 +201,8 @@ def pre_script(self):
             curr = curr + 1
 
         flag = True
-        for i in range(0,((self.timeoutInSeconds)/5)+2): #waiting 10 more seconds to escape race condition between Host and script timing out
-            time.sleep(5)
+        for i in range(0, int(self.timeoutInSeconds/self.pollTime) + 2): #waiting 10 more seconds to escape race condition between Host and script timing out
```
Comment (vityagi) [inline VMBackup/main/PluginHost.py]: make sure this syntax works in all like pyrthon 2.6
int(self.timeoutInSeconds/self.pollTime)

File: VMBackup/main/freezesnapshotter.py [hunk 2]
```diff
@@ -294,9 +294,10 @@ def takeSnapshotFromFirstGuestThenHost(self):
 
         run_result, run_status, blob_snapshot_info_array, all_failed, all_snapshots_failed, unable_to_sleep, is_inconsistent  = self.takeSnapshotFromGuest()
 
-        time.sleep(60) #sleeping for 60 seconds so that previous binary execution completes
-
         if(all_snapshots_failed):
+            #to make sure binary is thawed
+            self.logger.log('[takeSnapshotFromFirstGuestThenHost] : Thawing again post the guest snapshotting failure')
```
Comment (vityagi) [inline VMBackup/main/freezesnapshotter.py]: check if try-catch is needed here

File: VMBackup/main/PluginHost.py (hunk 1 above)
Comment (bpramod) [reply VMBackup/main/PluginHost.py]: both are made into int right...what exactly should i test ?

File: VMBackup/main/ScriptRunner.py [hunk 3]
```diff
@@ -102,6 +102,8 @@ def get_config(self):
                 self.postScriptNoOfRetries = configData['postScriptNoOfRetries']
             if 'fsFreezeEnabled' in configDataKeys:
                 self.fsFreeze_on = configData['fsFreezeEnabled']
+            if 'ScriptsExecutionPollTimeSeconds' in configDataKeys and int(configData['ScriptsExecutionPollTimeSeconds']) >= 1 and int(configData['ScriptsExecutionPollTimeSeconds']) <=5:
+                self.pollSleepTime = int(configData['ScriptsExecutionPollTimeSeconds'])
```
Comment (vityagi) [inline VMBackup/main/ScriptRunner.py]: add try-except ValueError to handle non-int case

Comment (bpramod) [reply VMBackup/main/ScriptRunner.py]: It is already in bigger try-catch.

File: VMBackup/main/freezesnapshotter.py [hunk 4]
```diff
@@ -294,9 +294,14 @@ def takeSnapshotFromFirstGuestThenHost(self):
...
 
-        time.sleep(60) #sleeping for 60 seconds so that previous binary execution completes
-
         if(all_snapshots_failed):
+            try:
+                #to make sure binary is thawed
+                self.logger.log('[takeSnapshotFromFirstGuestThenHost] : Thawing again post the guest snapshotting failure')
+                self.freezer.thaw_safe()
+            except Exception as e:
+                self.logger.log('[takeSnapshotFromFirstGuestThenHost] : Exception in Thaw - ' + str(e))
```
Comment (vityagi) [inline VMBackup/main/freezesnapshotter.py]: Trace in the way we trace exceptions with stack trace.
+ ': %s, stack trace: %s' % (str(err), traceback.format_exc())

Comment (bpramod) [reply VMBackup/main/freezesnapshotter.py]: Good Catch!

**Pull request merged**